adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -311,10 +313,22 @@ func (b *BM25Searcher) createTerm(N float64, query string, propertyNames []strin
 		if bucket == nil {
 			return termResult, nil, fmt.Errorf("could not find bucket for property %v", propName)
 		}
-		m, err := bucket.MapList([]byte(query))
+		preM, err := bucket.MapList([]byte(query))
 		if err != nil {
 			return termResult, nil, err
 		}
+		m := make([]lsmkv.MapPair, 0, len(preM))
```

Comment:
you could move this into the `if filterDocIds != nil` branch and only have the variable definition outside. Otherwise we allocate this when we do not need it

Reply:
@dirkkul I can quickly add this, then we can merge the PR.